src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergInitServiceIT.java

```diff
@@ -0,0 +1,74 @@
+package com.snowflake.kafka.connector.streaming.iceberg;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.snowflake.kafka.connector.internal.SnowflakeKafkaConnectorException;
+import com.snowflake.kafka.connector.internal.TestUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class IcebergInitServiceIT extends BaseIcebergIT {
+
+  private static IcebergInitService icebergInitService;
+
+  private String tableName;
+
+  @BeforeAll
+  // overrides the base class @BeforeAll
+  public static void setup() {
+    conn = TestUtils.getConnectionServiceForStreaming();
+    icebergInitService = new IcebergInitService(conn);
+  }
+
+  @BeforeEach
+  public void setUp() {
+    tableName = TestUtils.randomTableName();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    dropIcebergTable(tableName);
+  }
+
+  @Test
+  void shouldInitializeMetadataType() {
+    // given
+    createIcebergTable(tableName);
+
+    // when
+    icebergInitService.initializeIcebergTableProperties(tableName);
+
+    // then
+    assertThat(describeRecordMetadataType(tableName))
+        .isEqualTo(
+            "OBJECT(offset NUMBER(10,0), "
+                + "topic VARCHAR(16777216), "
+                + "partition NUMBER(10,0), "
+                + "key VARCHAR(16777216), "
+                + "schema_id NUMBER(10,0), "
+                + "key_schema_id NUMBER(10,0), "
+                + "CreateTime NUMBER(19,0), "
+                + "SnowflakeConnectorPushTime NUMBER(19,0), "
+                + "headers MAP(VARCHAR(16777216), "
+                + "VARCHAR(16777216)))");
+  }
+
+  @Test
+  void shouldThrowExceptionWhenTableDoesNotExist() {
+    assertThatThrownBy(() -> icebergInitService.initializeIcebergTableProperties(tableName))
+        .isInstanceOf(SnowflakeKafkaConnectorException.class);
+  }
+
+  @Test
+  void shouldThrowExceptionWhenRecordMetadataDoesNotExist() {
```

Comment:
I feel dumb. What is tested here?
We `ALTER ICEBERG TABLE identifier(?) ALTER COLUMN RECORD_METADATA ` in IcebergInitService... and also throw an error when there is no record_metadata ??

Reply:
By the way you can also verify exception code. Will be more readable.
```
    assertThatThrownBy(() -> JdbcProperties.create(connection, proxy, jdbcMap))
        .isInstanceOfSatisfying(
            SnowflakeKafkaConnectorException.class,
            ex -> {
              // property key is printed not value
              assertTrue(ex.getMessage().contains("user"));
              assertEquals("0031", ex.getCode());
            });
```

Reply:
```
net.snowflake.client.jdbc.SnowflakeSQLException: SQL compilation error: error line 1 at position 47
invalid identifier 'RECORD_METADATA'
```
I don't get it.

Reply:
aa I see we throw an error that there is no record_metadata column, when sql query throws an error...